pkg/controller/networkpolicy/validate.go

```diff
@@ -36,17 +65,45 @@ var (
 	// reservedTierNames stores the set of Tier names which cannot be deleted
 	// since they are created by Antrea.
 	reservedTierNames = sets.NewString("baseline", "application", "platform", "networkops", "securityops", "emergency")
+	// apv is an instance of antreaPolicyValidator to validate Antrea-native
+	// policy events.
+	apv = antreaPolicyValidator{}
+	// tv is an instance of tierValidator to validate Tier resource events.
+	tv = tierValidator{}
 )
 
+// RegisterAntreaPolicyValidator registers an Antrea-native policy validator to
+// the resource registry.
+func (v *NetworkPolicyValidator) RegisterAntreaPolicyValidator(a validator) {
+	v.antreaPolicyValidators = append(v.antreaPolicyValidators, a)
+}
+
+// RegisterTierValidator registers a Tier validator to the resource registry.
+func (v *NetworkPolicyValidator) RegisterTierValidator(t validator) {
+	v.tierValidators = append(v.tierValidators, t)
+}
```

Comment:
I would recommend specifying in the comments that these methods should be called before calling `Run` for the apiserver. Otherwise given that `antreaPolicyValidators` and `tierValidators` are not protected by a mutex, I image that there could be a race condition?

Reply:
done